aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -249,6 +254,81 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+    fun verify_key_rotation_signature(scheme: vector<u8>, public_key_bytes: vector<u8>, signature: vector<u8>, challenge: RotationProofChallenge) : vector<u8> {
+        let auth_key: vector<u8>;
+        if (scheme == ED25519_SCHEME) {
+            let pk = ed25519::new_unvalidated_public_key_from_bytes(public_key_bytes);
+            let sig = ed25519::new_signature_from_bytes(signature);
+            assert!(ed25519::signature_verify_strict_t(&sig, &pk, challenge), std::error::invalid_argument(EINVALID_PROOF_OF_KNOWLEDGE));
+            auth_key = ed25519::unvalidated_public_key_to_authentication_key(&pk);
+        } else {
+            let pk = multi_ed25519::new_unvalidated_public_key_from_bytes(public_key_bytes);
+            let sig = multi_ed25519::new_signature_from_bytes(signature);
+            assert!(multi_ed25519::signature_verify_strict_t(&sig, &pk, challenge), std::error::invalid_argument(EINVALID_PROOF_OF_KNOWLEDGE));
+            auth_key = multi_ed25519::unvalidated_public_key_to_authentication_key(&pk);
+        };
+        auth_key
+    }
+
+    fun get_challenge_message(account_address: address, in_scheme: vector<u8>, from_public_key_bytes: vector<u8>, to_public_key_bytes: vector<u8>): RotationProofChallenge acquires Account {
+        let account_resource = borrow_global_mut<Account>(account_address);
+
+        if (in_scheme == ED25519_SCHEME) {
+            let from_pk = ed25519::new_unvalidated_public_key_from_bytes(from_public_key_bytes);
+            let from_auth_key = ed25519::unvalidated_public_key_to_authentication_key(&from_pk);
+            assert!(account_resource.authentication_key == from_auth_key, error::unauthenticated(EWRONG_CURRENT_PUBLIC_KEY));
+        } else {
+            let from_pk = multi_ed25519::new_unvalidated_public_key_from_bytes(from_public_key_bytes);
+            let from_auth_key = multi_ed25519::unvalidated_public_key_to_authentication_key(&from_pk);
+            assert!(account_resource.authentication_key == from_auth_key, error::unauthenticated(EWRONG_CURRENT_PUBLIC_KEY));
+        };
+
+        let curr_auth_key = from_bcs::to_address(account_resource.authentication_key);
+
+        let challenge = RotationProofChallenge {
+            sequence_number: account_resource.sequence_number,
+            originator: account_address,
+            current_auth_key: curr_auth_key,
+            new_public_key: to_public_key_bytes,
+        };
+
+        challenge
+    }
+
+    public entry fun rotate_authentication_key(
+        account: &signer,
+        in_scheme: vector<u8>,
+        to_scheme: vector<u8>,
+        from_public_key_bytes: vector<u8>,
+        to_public_key_bytes: vector<u8>,
```

Comment:
minor nit... order from_scheme, from_bytes, to_scheme, to_bytes
and keep consistent naming
that will make it easier for most users since names and ordering map flow

Reply:
Yes, can we use consistent naming: e.g., use "from_scheme" (rather than "in_scheme")